<?xml version="1.0" ?>
<launch>

	<!-- GENERAL ARGUMENTS -->
	<arg name="robot_name" default="panda"/>
	<arg name="tool_name" default="microGRAS"/>
	<arg name="simulation" default="false"/>
	<arg name="calibration" default="true"/>
	<arg name="safety_distance" default="0.04"/>

	<!-- ROBOT SETUP -->
	<include file="$(find robot_arm_tools)/launch/Setup_$(arg robot_name).launch">
		<arg name="tool_name" value="$(arg tool_name)"/>
		<arg name="simulation" value="$(arg simulation)"/>
		<arg name="calibration" value="$(arg calibration)"/>
		<arg name="safety_distance" value="$(arg safety_distance)"/>
	</include>

	<!-- STUDIED OBJECT -->
	<node pkg="robot_arm_tools" type="robot_arm_tools_environment_node" name="robot_arm_tools_environment_node_object" output="screen">
		<param name="environmentFileName" value="$(find robot_arm_acoustic)/config/environments/StudiedObject.yaml"/>
	</node>

	<!-- MEASUREMENT SERVER -->
	<arg name="measurement_server_storage_folder" default="/home/demo1/AcousticMeasurements/Verification/"/>
	<param name="measurementServerName" value="/sound_measurement_server"/>
	<param name="measurementServerStorageFolder" value="$(arg measurement_server_storage_folder)"/>
	<node pkg="robot_arm_acoustic" type="SoundMeasurementServer.py" name="sound_measurement_server" output="screen"/>

	<!-- MAIN NODE -->
	<arg name="trajectory_radius" default="0.5"/>
	<arg name="trajectory_steps_number" default="10"/>
	<arg name="trajectory_axis" default="[0,0,1]"/>
	<arg name="trajectory_center_pose" default="[0,0,0,0,0,0]"/>
	
	<param name="trajectoryRadius" value="$(arg trajectory_radius)"/>
	<param name="trajectoryStepsNumber" value="$(arg trajectory_steps_number)"/>
	<param name="trajectoryAxis" type="yaml" value="$(arg trajectory_axis)"/>
	<param name="trajectoryCenterPose" type="yaml" value="$(arg trajectory_center_pose)"/>
	
	<node pkg="robot_arm_acoustic" type="robot_arm_acoustic_directivity_measurement_node" name="robot_arm_acoustic_directivity_measurement_node" output="screen" required="true"/>

	<!-- TO REMOVE -->
	<arg name="gradient" default="false"/>
	<arg name="gradient_offset"	default="0.01"/>
	<param name="gradientMeasurements" value="$(arg gradient)"/>
	<param name="gradientMeasurementsOffset" value="$(arg gradient_offset)"/>

	<!-- Run Rviz --> 
    <node pkg="robot_arm_tools" type="robot_arm_tools_environment_node" name="robot_arm_tools_environment_node_environment" output="screen">
		<param name="environmentFileName" value="$(find robot_arm_acoustic)/config/environments/GarteurAnechoicRoom.yaml"/>
	</node>
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find robot_arm_acoustic)/config/AcousticSetup.rviz"/>

</launch>
